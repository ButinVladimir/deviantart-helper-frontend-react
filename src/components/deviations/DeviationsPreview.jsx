import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { DEVIATIONS_DETAILS } from '../../consts/routes';

export default function DeviationsPreview({
  thumbnail,
  title,
  url,
  id,
  publishedTime,
  views,
  favourites,
  comments,
  downloads,
}) {
  return (
    <li>
      <div>
        <img alt={title} src={thumbnail.src} width={thumbnail.width} height={thumbnail.height} />
      </div>
      <div>
        <span>{title}</span>
        <span> (</span>
        <a href={url}>See on DA</a>
        <span>)</span>
        <span> (</span>
        <NavLink to={`${DEVIATIONS_DETAILS}${id}`}>See details</NavLink>
        <span>)</span>
      </div>
      <div>
        <span>Id: </span>
        <span>{id}</span>
      </div>
      <div>
        <span>Published time: </span>
        <span>{new Date(publishedTime).toLocaleString()}</span>
      </div>
      <div>
        <span>Views: </span>
        <span>{views}</span>
      </div>
      <div>
        <span>Favourites: </span>
        <span>{favourites}</span>
      </div>
      <div>
        <span>Comments: </span>
        <span>{comments}</span>
      </div>
      <div>
        <span>Downloads: </span>
        <span>{downloads}</span>
      </div>
    </li>
  );
}

DeviationsPreview.propTypes = {
  thumbnail: PropTypes.shape({
    src: PropTypes.string.isRequired,
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
  }).isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  publishedTime: PropTypes.number.isRequired,
  views: PropTypes.number.isRequired,
  favourites: PropTypes.number.isRequired,
  comments: PropTypes.number.isRequired,
  downloads: PropTypes.number.isRequired,
};
