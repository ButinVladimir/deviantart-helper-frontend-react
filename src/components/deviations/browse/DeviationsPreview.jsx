import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import Media from 'react-bulma-components/lib/components/media';
import Content from 'react-bulma-components/lib/components/content';
import Heading from 'react-bulma-components/lib/components/heading';
import DeviationThumbnail from '../shared/DeviationThumbnail';
import StatisticsLevel from '../../shared/StatisticsLevel';
import { DEVIATIONS_DETAILS, ID_PARAM } from '../../../consts/routes';

export default function DeviationsPreview({
  thumbnail,
  title,
  url,
  id,
  publishedTime,
  nsfw,
  views,
  favourites,
  comments,
  downloads,
}) {
  return (
    <Media>
      <DeviationThumbnail
        title={title}
        thumbnail={thumbnail}
        url={url}
        publishedTime={publishedTime}
        nsfw={nsfw}
      />
      <Media.Item>
        <Content>
          <Heading size={3}>
            <NavLink to={`${DEVIATIONS_DETAILS.replace(ID_PARAM, id)}`}>
              {title}
            </NavLink>
          </Heading>
        </Content>
        <StatisticsLevel
          titleValuePairs={[['Views', views], ['Favourites', favourites], ['Comments', comments], ['Downloads', downloads]]}
        />
      </Media.Item>
    </Media>
  );
}

DeviationsPreview.propTypes = {
  ...DeviationThumbnail.PropTypes,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  views: PropTypes.number.isRequired,
  favourites: PropTypes.number.isRequired,
  comments: PropTypes.number.isRequired,
  downloads: PropTypes.number.isRequired,
};
