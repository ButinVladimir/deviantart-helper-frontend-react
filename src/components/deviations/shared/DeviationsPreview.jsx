import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import Media from 'react-bulma-components/lib/components/media';
import Content from 'react-bulma-components/lib/components/content';
import Heading from 'react-bulma-components/lib/components/heading';
import { Field, Control, Checkbox } from 'react-bulma-components/lib/components/form';
import DeviationThumbnail from './DeviationThumbnail';
import StatisticsLevel from '../../shared/StatisticsLevel';
import { DEVIATIONS_DETAILS, ID_PARAM } from '../../../consts/routes';

export default function DeviationsPreview({
  thumbnail,
  title,
  url,
  id,
  publishedTime,
  nsfw,
  views,
  favourites,
  comments,
  downloads,
  selected,
  toggleSelectionHandler,
}) {
  return (
    <Media>
      <DeviationThumbnail
        title={title}
        thumbnail={thumbnail}
        url={url}
        publishedTime={publishedTime}
        nsfw={nsfw}
      />
      <Media.Item>
        <Content>
          <Heading size={3}>
            <NavLink to={`${DEVIATIONS_DETAILS.replace(ID_PARAM, id)}`}>
              {title}
            </NavLink>
          </Heading>
        </Content>
        <StatisticsLevel
          titleValuePairs={[['Views', views], ['Favourites', favourites], ['Comments', comments], ['Downloads', downloads]]}
        />
      </Media.Item>
      <Media.Item position="right">
        <Field>
          <Control>
            <Checkbox checked={selected} onChange={() => toggleSelectionHandler(id)} />
          </Control>
        </Field>
      </Media.Item>
    </Media>
  );
}

DeviationsPreview.propTypes = {
  ...DeviationThumbnail.PropTypes,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  views: PropTypes.number.isRequired,
  favourites: PropTypes.number.isRequired,
  comments: PropTypes.number.isRequired,
  downloads: PropTypes.number.isRequired,
  selected: PropTypes.bool.isRequired,
  toggleSelectionHandler: PropTypes.func.isRequired,
};
