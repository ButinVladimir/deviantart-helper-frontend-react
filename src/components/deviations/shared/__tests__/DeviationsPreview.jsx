import React from 'react';
import { shallow } from 'enzyme';
import DeviationsPreview from '../DeviationsPreview';

describe('DeviationsPreview', () => {
  const deviationProps = {
    id: '1',
    title: 'Title',
    thumbnail: {
      src: '/path/to/thumbnail',
      width: 1,
      height: 2,
    },
    nsfw: true,
    url: '/path/to/deviation',
    publishedTime: 1234,
    views: 1,
    favourites: 2,
    comments: 3,
    downloads: 4,
    selected: true,
  };

  it('can be rendered correctly', () => {
    const wrapper = shallow(
      <DeviationsPreview {...deviationProps} toggleSelectionHandler={() => {}} />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('can handle toggling selection', () => {
    const toggleSelectionHandler = jest.fn();
    const wrapper = shallow(
      <DeviationsPreview {...deviationProps} toggleSelectionHandler={toggleSelectionHandler} />,
    );

    wrapper.find('#selection-1').simulate('change');

    expect(toggleSelectionHandler).toHaveBeenCalled();
  });
});
