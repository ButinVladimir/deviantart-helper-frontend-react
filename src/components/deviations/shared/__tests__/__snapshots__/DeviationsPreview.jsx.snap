// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviationsPreview can be rendered correctly 1`] = `
<Media
  className=""
  renderAs="article"
  style={Object {}}
>
  <DeviationThumbnail
    nsfw={true}
    publishedTime={1234}
    thumbnail={
      Object {
        "height": 2,
        "src": "/path/to/thumbnail",
        "width": 1,
      }
    }
    title="Title"
    url="/path/to/deviation"
  />
  <MediaItem
    className=""
    position="center"
    renderAs="div"
    style={Object {}}
  >
    <Content
      renderAs="div"
      size={null}
      style={Object {}}
    >
      <Heading
        className=""
        heading={false}
        renderAs="h1"
        size={3}
        spaced={false}
        subtitle={false}
        weight={null}
      >
        <NavLink
          to="/deviations/details/1"
        >
          Title
        </NavLink>
      </Heading>
    </Content>
    <StatisticsLevel
      titleValuePairs={
        Array [
          Array [
            "Views",
            1,
          ],
          Array [
            "Favourites",
            2,
          ],
          Array [
            "Comments",
            3,
          ],
          Array [
            "Downloads",
            4,
          ],
        ]
      }
    />
  </MediaItem>
  <MediaItem
    className=""
    position="right"
    renderAs="div"
    style={Object {}}
  >
    <Field
      align={null}
      className=""
      horizontal={false}
      kind={null}
      multiline={false}
      renderAs="div"
      style={Object {}}
    >
      <Control
        className=""
        fullwidth={false}
        iconLeft={false}
        iconRight={false}
        loading={false}
        renderAs="div"
        size={null}
        style={Object {}}
      >
        <Checkbox
          checked={true}
          className=""
          disabled={false}
          id="selection-1"
          name={null}
          onChange={[Function]}
          style={Object {}}
          value=""
        />
      </Control>
    </Field>
  </MediaItem>
</Media>
`;
